lib/cover.nix

```diff
@@ -0,0 +1,98 @@
+{ stdenv, lib, haskellLib, pkgs }:
+
+{ name, version, library, tests, plan }:
+
+let
+  buildWithCoverage = builtins.map (d: d.covered);
+  runCheck = builtins.map (d: haskellLib.check d);
+
+  libraryCovered    = library.covered;
+  testsWithCoverage = buildWithCoverage tests;
+  checks            = runCheck testsWithCoverage;
+
+  identifier = name + "-" + version;
+
+  getTestModulesFor = test: plan.components.tests."${test.exeName}".modules;
+
+in stdenv.mkDerivation {
+  name = (identifier + "-coverage-report");
+
+  phases = ["buildPhase"];
+
+  buildInputs = (with pkgs; [ ghc ]);
+
+  buildPhase = ''
+    findMixDir() {
+      find $1 -iwholename "*/hpc/vanilla/mix" -exec find {} -maxdepth 1 -type d -iwholename "*/mix/*" \; -quit
+    }
+
+    findCabalFile() {
```

Comment:
dead since the cabal file var is dead

Reply:
Addressed https://github.com/input-output-hk/haskell.nix/pull/762/commits/0f8010ca7979d258a0c5fd3dd7e48f88f5817aa1